development/testing/testing_pull_request.py

```diff
@@ -1,88 +1,91 @@
+from pathlib import Path
 
-CURRENT_DIR = os.path.dirname(os.path.realpath(__file__))
-for dirname in ["regression", "property", "release", "robustness", "parallelism"]:
-    sys.path.insert(0, CURRENT_DIR + "/" + dirname)
-
-from run_regression import run as run_regression
-from run_property import run as run_property
-from run_robustness import run as run_robustness
-from run_parallelism import run as run_parallelism
-
-# Here we specify the group of tests to run. Later we also pin down the details.
-request_dict = dict()
-request_dict["REGRESSION"] = True
-request_dict["PROPERTY"] = True
-request_dict["PYTEST"] = True
-request_dict["ROBUSTNESS"] = True
-request_dict["PARALLELISM"] = True
-
-# determine whether to do a long or short run
-short_run = gethostname() in ["socrates"]
-
-# We need to specify the arguments for each of the tests.
-test_spec = dict()
-test_spec["PYTEST"] = dict()
-
-test_spec["REGRESSION"] = dict()
-if short_run is True:
-    test_spec["REGRESSION"]["request"] = ("check", 200)
-else:
-    test_spec["REGRESSION"]["request"] = ("check", 10000)
-test_spec["REGRESSION"]["is_background"] = False
-test_spec["REGRESSION"]["is_compile"] = False
-test_spec["REGRESSION"]["is_strict"] = True
-test_spec["REGRESSION"]["num_procs"] = 3
-
-test_spec["PROPERTY"] = dict()
-if short_run is True:
-    test_spec["PROPERTY"]["request"] = ("run", 0.1)
-else:
-    test_spec["PROPERTY"]["request"] = ("run", 12)
-test_spec["PROPERTY"]["is_background"] = False
-test_spec["PROPERTY"]["is_compile"] = False
-
-test_spec["ROBUSTNESS"] = dict()
-if short_run is True:
-    test_spec["ROBUSTNESS"]["request"] = ("run", 0.1)
-else:
-    test_spec["ROBUSTNESS"]["request"] = ("run", 12)
-test_spec["ROBUSTNESS"]["is_compile"] = False
-test_spec["ROBUSTNESS"]["is_background"] = False
-test_spec["ROBUSTNESS"]["keep_dataset"] = False
-test_spec["ROBUSTNESS"]["num_procs"] = 3
-
-test_spec["PARALLELISM"] = dict()
-if short_run is True:
-    test_spec["PARALLELISM"]["hours"] = 0.1
-else:
-    test_spec["PARALLELISM"]["hours"] = 12
-
-
-if request_dict["PYTEST"]:
-    respy.test()
-
-if request_dict["REGRESSION"]:
-    os.chdir("regression")
-    run_regression(**test_spec["REGRESSION"])
-    os.chdir(CURRENT_DIR)
-
-if request_dict["PROPERTY"]:
-    os.chdir("property")
-    run_property(**test_spec["PROPERTY"])
-    os.chdir(CURRENT_DIR)
-
-if request_dict["ROBUSTNESS"]:
-    os.chdir("robustness")
-    run_robustness(**test_spec["ROBUSTNESS"])
-    os.chdir(CURRENT_DIR)
-
-if request_dict["PARALLELISM"]:
-    os.chdir("parallelism")
-    run_parallelism(**test_spec["PARALLELISM"])
-    os.chdir(CURRENT_DIR)
+from development.testing.regression.run_regression import run as run_regression
+from development.testing.property.run_property import run as run_property
+from development.testing.robustness.run_robustness import run as run_robustness
+from development.testing.parallelism.run_parallelism import run as run_parallelism
+
+CURRENT_DIR = Path(__file__).parent
```

Comment:
This script is not running properly anymore due to the revised creation of `CURRENT_DIR`. Please fix so I can run it a bit ....

Reply:
I think I fixed it. Please check again.

Reply:
The problem persists as CURRENT_DIR is just '.' and this does not suffice to return to the proper directory level. Please just test by commenting out all testing but just the stepping in and out of the testing subdirectories.

Reply:
But now :). Sorry for not checking again. The path was relative, but to return to the previous directory we needed the absolute path.

Reply:
This requires further adjustments and not just replacing '+' by '/'. The package dir was usually referring to the package root but this does not appear to be the case here anymore .....
   https://github.com/OpenSourceEconomics/respy/blob/b9efe5f88eaf2061d5aa99b02dfc03bc6750acf8/development/testing/property/run_property.py#L55 


`base) peisenha@pontos:~/.../development/testing$ python testing_pull_request.py 
Traceback (most recent call last):
  File "testing_pull_request.py", line 91, in <module>
    main()
  File "testing_pull_request.py", line 76, in main
    run_property(**test_spec["PROPERTY"])
  File "/home/peisenha/OpenSourceEconomics/respy/development/testing/property/run_property.py", line 55, in run
    test_dict = get_test_dict(PACKAGE_DIR / "respy/tests")
  File "/home/peisenha/OpenSourceEconomics/respy/development/modules/auxiliary_property.py", line 146, in get_test_dict
    os.chdir(test_dir)
FileNotFoundError: [Errno 2] No such file or directory: '/home/peisenha/OpenSourceEconomics/respy/development/testing/property/development/testing/property/respy/tests'
`

Reply:
Please be sure to address #161 before my next review.